warehouse/oidc/models/github.py

```diff
@@ -205,6 +210,10 @@ def job_workflow_ref(self):
     def sub(self):
         return f"repo:{self.repository}"
 
+    @property
+    def ref(self):
+        return unpredictable
```

Comment:
Now that the `check` callable is optional, this can be removed, right? If I'm reading the validation logic correctly, we should never access it.

Reply:
Probably, although I'd need to rework some tests that expect all required claims to have properties on the publisher.

Reply:
Ah yeah, maybe not worth it then.

Reply:
I don't understand the sentinel `unpredictable` here, is it just to return an object that is not `None`?

Reply:
Yeah, I think that's mostly to prevent someone from accidentally comparing on it.

Reply:
What @woodruffw said, but I decided that it'd be preferable for this property to just not exist instead: 
6a3ddfa